### ISSUE TITLE
On the git version being used in the docker.

### ISSUE THREAD

**Issue by fermadeiral** [Spirals-Team/repairnator#769]
We do not set or install a git version on the docker. It might be good if we do this, to be able to choose a version and to be sure in which version we are running git commands.

**Comment by fermadeiral** [comment]
> Why would you need a specific version of git?

I don't really need a specific git version. But few months ago I run the docker with a newer git version compared to the one that is currently taken (I don't remember which is the version, but it's a little bit old...I need to check it again), and I got some different results. In that time, I didn't search for the root cause, i.e. which git command changed the behaviour, but my point is that there was a difference between the versions. I opened this issue having in mind that maybe we could take a newer version, or at least to make explicity in the docker file which version we are using now.

Currently we do not have too much git commands in Repairnator (that we call directly), but one day when issue https://github.com/Spirals-Team/repairnator/issues/361 is fixed, it might be better for us to look closer to the git version we use.

**Comment by surli** [comment]
We rely on tagged docker images, so the Git version should not change if we use the same base docker image everywhere.

Now I follow you on the need to know the git version for Debugging purpose. What thing we can easily do is to print the Git version in first lines of the log.

**Comment by fermadeiral** [comment]
> We rely on tagged docker images, so the Git version should not change if we use the same base docker image everywhere.

I agree. Just to make sure that I didn't pass the wrong idea: when I run the docker with a different git version was because I forced it to happen, i.e. I do not claim that the git version change by itself.

> Now I follow you on the need to know the git version for Debugging purpose. What thing we can easily do is to print the Git version in first lines of the log.

Good, I would like that a lot. Again, I do not need a specific git version (at least I never had a reason for it so far), and making explicit for us to see which git version we are using will be peaceful to me.
